### PR TITLE
'Update' page frontend

### LINKED ISSUE #74: Frontend for /update page
https://github.com/comicrelief/serverless-giftaid/issues/18

Design is ready!
Interactive prototype: https://50cr8l.axshare.com (where you see it can be clicked there is an interaction, otherwise refer to the current gift aid journey interaction)

Visual design is attached here and found in: CR product development > Gift Aid > Gift Aid - user journey.sketch or CR product development > Gift Aid > Edit declaration

image
image
image
image

### PR DESCRIPTION
**Pull request by AndyEPhipps**
Fixes #74

### CODE REVIEW

File: src/pages/UpdateForm/UpdateForm.js [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+    }
+    clearTimeout(scrollTimeout);
+  }
+
+  /**
+   * Maps the input field properties to a new array containing the input field instances
+   * @returns {Array}
+   */
+  createInputFields() {
+    const allFields = this.state.inputFieldProps;
```
Comment (Heleen-cr) [inline src/pages/UpdateForm/UpdateForm.js]: I think there's no need to use the state for this anymore as we can directly use the `defaultInputFieldsData` import right?

File: src/pages/UpdateSuccess/UpdateSuccess.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    /*    const hasState = this.props.location.state;
+    const firstName = this.props.location.state.firstname;
+    const giftAidChoice = this.props.location.state.giftAidChoice; */
+
+    // HARDCODED DEBUG VALUES!
+    const hasState = true;
+    const firstName = 'Andy';
+    let giftAidChoice = 0;
+    giftAidChoice = 1;
+
```
Comment (Heleen-cr) [inline src/pages/UpdateSuccess/UpdateSuccess.js]: not sure if you're finished, so just a reminder that these are still there. (And the commented out properties at the bottom.)

File: src/pages/UpdateForm/UpdateForm.js (hunk 1 above)
Comment (AndyEPhipps) [reply src/pages/UpdateForm/UpdateForm.js]: Good point

File: src/pages/UpdateSuccess/UpdateSuccess.js (hunk 2 above)
Comment (AndyEPhipps) [reply src/pages/UpdateSuccess/UpdateSuccess.js]: Will be doing the success stuff in the next PR, as discussed

**Pull request merged**